### ISSUE TITLE
🐞[BUG]: Cannot assign @Select to private property

### ISSUE THREAD

**Issue by twjacobsen** [ngxs/store#1532]
### Affected Package

 The issue is caused by package @ngxs/5.6.2

### Is this a regression?

 Yes, it worked just fine in 3.4.3

### Description

 A clear and concise description of the problem...
Using the `@Select` decorator to assign to private properties of another class results in this error:
```
Argument of type 'HelloComponent' is not assignable to parameter of type 'ComponentClass<any> & Record<"_name", Observable<string>>'.
  Type 'HelloComponent' is not assignable to type 'Record<"_name", Observable<string>>'.
    Property '_name' is private in type 'HelloComponent' but not in type 'Record<"_name", Observable<string>>'.
```

## 🔬 Minimal Reproduction

 
https://stackblitz.com/edit/ngxs-at-select-private-property?file=src%2Fapp%2Fhello.component.ts

## Environment

<pre><code>
Libs:
- @angular/core version: 8.2.4
- @ngxs/store version: 5.6.2
- Typescript version: 3.7.5

**Comment by splincode** [comment]
please create stackblitz or guthub repo with reproduce

**Comment by twjacobsen** [comment]
There is a link under "Minimal reproduction"... IDK why it won't make it an actual link..

**Comment by markwhitfeld** [comment]
@splincode I believe that this would have been caused by the extra typings you added in this commit:
https://github.com/ngxs/store/pull/1436/commits/ca7de82b3032057743dbd1cb4e50a5cf934011d1
Are you able to improve the types to handle private fields?

**Comment by splincode** [comment]
Sorry, I read the description poorly and did not look at the link

**Comment by splincode** [comment]
Sorry, I read the description poorly and did not look at the link.

@markwhitfeld I will investigate

**Comment by steve-todorov** [comment]
I'm getting this issue as well. In our case we have an `AbstractAuthorityGuard` which has a "default" implementation of `CanActivate` and allows for other `Guard`s to extend the abstract class and do minor additional checks. Sort of like this oversimplified example:

```
@Injectable({
    providedIn: 'root'
})
export abstract class AbstractAuthorityGuard implements CanActivate {
    @Select(SessionState.authorities)
    protected authorities$: Observable<UserAuthority[]>;

    canActivate(): Observable<boolean> {
        return this.authorities$;
    }
}
```

**Comment by steve-todorov** [comment]
Hey guys, any update on this? :)

**Comment by splincode** [comment]
While this is a difficult time, I will get to this problem soon

I have now completely concentrated my efforts on a new plugin that will improve working with states:
https://github.com/ngxs-labs/data

**Comment by steve-todorov** [comment]
Thanks for the update! `ngxs-labs/data` looks very promising! Is there a release planned for the near future?

**Comment by splincode** [comment]
It is planned to implement a lot of functionality yet, but I hope to make a release at the end of this month

**Comment by bschantz** [comment]
Any update? I've got a large codebase that I'd like to upgrade to Angular 9, but this issue is blocking.

**Comment by markwhitfeld** [comment]
It looks like @splincode is working on this in PR #1623. Hope to get this fix into the next release!

**Comment by steve-todorov** [comment]
It looks like you guys have merged the PR already. Is there anything more that needs to be done for this to get released? :) 

Off-topic: How are things going with `ngxs-labs/data`?

**Comment by markwhitfeld** [comment]
Good news this has been addressed in NGXS v3.7, released last week.
( see https://dev.to/ngxs/announcing-ngxs-3-7-1kbk for release notes )

I will close this issue. Please let us know if it is not resolved.

**Comment by steve-todorov** [comment]
Awesome news! Thanks! :)
